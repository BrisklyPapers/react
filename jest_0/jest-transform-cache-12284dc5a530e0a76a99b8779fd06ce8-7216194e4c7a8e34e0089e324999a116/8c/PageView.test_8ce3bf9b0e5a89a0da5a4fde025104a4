'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _PageView = require('../../src/components/PageView.js');var _PageView2 = _interopRequireDefault(_PageView);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/PageView', function () {
    it('renders correctly', function () {
        var pageView = _reactTestRenderer2.default.create(
        _react2.default.createElement(_PageView2.default, { visible: true }, _react2.default.createElement('span', null, 'Child')));

        expect(pageView.toJSON()).toMatchSnapshot();
    });

    it('renders without children', function () {
        var pageView = _reactTestRenderer2.default.create(
        _react2.default.createElement(_PageView2.default, { visible: false }, _react2.default.createElement('span', null, 'Child')));

        expect(pageView.toJSON()).toMatchSnapshot();
    });
});