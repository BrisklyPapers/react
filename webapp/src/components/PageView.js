import React from 'react'
import PropTypes from 'prop-types'

class PageView extends React.Component {
    render() {
        return (
        <div>
        {this.props.visible &&
            this.props.children
        }
        </div>
        )
    }
}

PageView.propTypes = {
    visible: PropTypes.bool.isRequired
};

export default PageView;