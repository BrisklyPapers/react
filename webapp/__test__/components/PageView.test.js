import React from 'react';
import PageView from '../../src/components/PageView.js';
import renderer from 'react-test-renderer';

describe('components/PageView', () => {
    it('renders correctly', () => {
        const pageView = renderer.create(
            <PageView visible={true}><span>Child</span></PageView>
        );
        expect(pageView.toJSON()).toMatchSnapshot();
    });
});