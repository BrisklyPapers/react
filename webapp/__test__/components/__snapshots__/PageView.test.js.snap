// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PageView renders correctly 1`] = `
<div>
  <span>
    Child
  </span>
</div>
`;
